tools/harness-automation/autothreadharness/runner.py

```diff
@@ -143,20 +144,36 @@ def addError(self, test, err):
         super(SimpleTestResult, self).addError(test, err)
         self.add_result(test, None, str(err[1]))
 
+
+def __print_fw_version_of_device_connected_to_port(port):
+    """Print firmware version of a device connected to the COM port"""
+    try:
+        with OpenThreadController(port) as otc:
+            print('%s: %s' % (port, otc.version))
+    except:
+        logger.exception('failed to get version of %s' % port)
+
+
 def list_devices(names=None, continue_from=None, **kwargs):
     """List devices in settings file and print versions"""
 
-    if continue_from:
-        continue_from = settings.GOLDEN_DEVICES.index(continue_from)
+    ports = []
+
+    if names:
+        ports = list(names)
+
     else:
-        continue_from = 0
+        if continue_from:
+            continue_from = settings.GOLDEN_DEVICES.index(continue_from)
+        else:
+            continue_from = 0
+
+        devices = list(settings.GOLDEN_DEVICES[continue_from:])
+
+        ports = [port for port, _type in devices if _type == 'OpenThread']
```

Comment:
I think this filter and conversion should be done before `continue_from` processing.  `continue_from` is the port name only.